Library/ViewModels/ProjectCreatorViewModelTests.swift

```diff
@@ -65,6 +117,47 @@ final class ProjectCreatorViewModelTests: TestCase {
     self.goToMessageDialogSubject.assertValues([.project(project)])
   }
 
+  func testGoToMessageDialog_LoggedOut() {
+    let project = Project.template
+    self.vm.inputs.configureWith(project: project)
+    self.vm.inputs.viewDidLoad()
+
+    self.goToMessageDialogContext.assertValues([])
+    self.goToMessageDialogSubject.assertValues([])
+
+    let creatorBioRequest = URLRequest(
+      url: URL(string: "https://www.kickstarter.com/projects/a/b/creator_bio")!
+    )
+    var policy = self.vm.inputs.decidePolicy(
+      forNavigationAction: WKNavigationActionData(
+        navigationType: .other,
+        request: creatorBioRequest,
+        sourceFrame: WKFrameInfoData(mainFrame: true, request: creatorBioRequest),
+        targetFrame: WKFrameInfoData(mainFrame: true, request: creatorBioRequest)
+      )
+    )
+    XCTAssertEqual(.allow, policy)
+
+    self.goToMessageDialogContext.assertValues([])
+    self.goToMessageDialogSubject.assertValues([])
+
+    let messagesRequest = URLRequest(
+      url: URL(string: "https://www.kickstarter.com/projects/a/b/messages/new")!
+    )
+    policy = self.vm.inputs.decidePolicy(
+      forNavigationAction: WKNavigationActionData(
+        navigationType: .linkActivated,
+        request: messagesRequest,
+        sourceFrame: WKFrameInfoData(mainFrame: true, request: messagesRequest),
+        targetFrame: WKFrameInfoData(mainFrame: true, request: messagesRequest)
+      )
+    )
+    XCTAssertEqual(.cancel, policy)
+
+    self.goToMessageDialogContext.assertValues([])
+    self.goToMessageDialogSubject.assertValues([])
```

Comment:
For simplicity, you could just test this output in your `testGoToLoginTout_LoggedOut` function and just add this one line to show that the message dialog isn't opened. But no bigs if this separate function feels more clear to you.